tests/Unit/MerchantCenter/PolicyComplianceCheckTest.php

```diff
@@ -0,0 +1,138 @@
+		$this->wc->expects( $this->any() )
+					   ->method( 'get_allowed_countries' )
+					->willReturn(
+						[
+							'AU' => 'Australia',
+							'AT' => 'Austria',
+							'CA' => 'Canada',
+							'US' => 'United States',
+						]
+					);
+		$this->target_audience->expects( $this->any() )
+					   ->method( 'get_target_countries' )
+					   ->willReturn( [ 'AU', 'US' ] );
+
+		$this->assertTrue( $this->policy_compliance_check->is_accessible() );
+	}
+
+	public function test_website_not_accessible() {
+		$this->wc->expects( $this->any() )
+					   ->method( 'get_allowed_countries' )
+					->willReturn(
+						[
+							'AU' => 'Australia',
+							'AT' => 'Austria',
+							'CA' => 'Canada',
+							'US' => 'United States',
+						]
+					);
+		$this->target_audience->expects( $this->any() )
+					   ->method( 'get_target_countries' )
+					   ->willReturn( [ 'FR', 'US' ] );
+
+		$this->assertFalse( $this->policy_compliance_check->is_accessible() );
+	}
+
+	public function test_payment_gateways() {
+		$this->wc->expects( $this->any() )
+					   ->method( 'get_available_payment_gateways' )
+					->willReturn(
+						[
+							'stripe' => $this->createMock( WC_Payment_Gateway::class ),
+							'paypal' => $this->createMock( WC_Payment_Gateway::class ),
+						]
+					);
+
+		$this->assertTrue( $this->policy_compliance_check->has_payment_gateways() );
+	}
+
+	public function test_empty_payment_gateways() {
+		$this->wc->expects( $this->any() )
+					   ->method( 'get_available_payment_gateways' )
+					   ->willReturn( [] );
+
+		$this->assertFalse( $this->policy_compliance_check->has_payment_gateways() );
+	}
+
+	public function test_not_has_page_not_found_error() {
+		$this->assertFalse( $this->policy_compliance_check->has_page_not_found_error() );
+	}
+
+	public function test_has_page_not_found_error() {
+		$this->mock_wp_request( 'http://example.org', '', 404 );
+		$this->assertTrue( $this->policy_compliance_check->has_page_not_found_error() );
+	}
+
+	public function test_not_has_redirects() {
+		$this->assertFalse( $this->policy_compliance_check->has_redirects() );
+	}
+
+	public function test_has_redirects() {
+		$this->mock_wp_request( 'http://example.org', '', 301 );
+		$this->assertTrue( $this->policy_compliance_check->has_redirects() );
+	}
+
+	public function test_not_has_restrictions() {
+		$this->assertFalse( $this->policy_compliance_check->has_restriction() );
+	}
+
+	public function test_has_restrictions() {
+		$this->mock_wp_request( 'http://example.org/robots.txt', "User-agent: *\nDisallow: /" );
+		$this->assertTrue( $this->policy_compliance_check->has_restriction() );
+	}
+
+	public function test_without_store_ssl() {
+		$this->assertFalse( $this->policy_compliance_check->get_is_store_ssl() );
+	}
+
+	public function test_with_store_ssl() {
+		define( 'WP_HOME', 'https://example.org' );
+		$this->assertTrue( $this->policy_compliance_check->get_is_store_ssl() );
+	}
+
+	public function test_not_has_refund_page() {
```

Comment:
It would be nice to test for an existing refund page as well.

Something like this should work:

```php
	public function test_has_refund_page() {
		$page_id = wp_insert_post(
			[
				'post_title'  => 'Refund and Returns',
				'post_name'   => 'refund_returns',
				'post_type'   => 'page',
				'post_status' => 'publish',
			]
		);

		$this->assertTrue( $this->policy_compliance_check->has_refund_return_policy_page() );
		wp_delete_post( $page_id, true );
	}
```